### PR TITLE
Removed Product Collection Unique ID

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

With the availability of other unique characteristics for identifying collection blocks added in https://github.com/woocommerce/woocommerce/pull/43997, it is no longer necessary to keep this attribute. Since it has not been used it is safe to remove it without a deprecation cycle.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

N/A

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Removed `id` attribute from Product Collection blocks.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx
Comment (imanish003) [inline plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx:1]: I think you forgot to remove the import statement:
```tsx
import { v4 as uuidV4 } from 'uuid';
```

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/block.json
Comment (imanish003) [inline plugins/woocommerce-blocks/assets/js/blocks/product-collection/block.json:1]: I think you will also need to remove "id" from usesContext [here](https://github.com/woocommerce/woocommerce/blob/bee675c66ca75315482590140fe23a47ab4d4bde/plugins/woocommerce-blocks/assets/js/blocks/product-template/block.json#L11)

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx
Comment (imanish003) [inline plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx:1]: Considering we initially added `uuid` and `@types/uuid` to `package.json` upon introducing "id", and now that they're no longer needed, it seems logical to remove them as well. What are your thoughts on this?

**Review by imanish003 — CHANGES_REQUESTED** [review]
Thanks for removing the "id". I left a few minor comments 🙂

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx
Comment (ObliviousHarmony) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx:1]: Sorry, yeah! This PR was extracted from my other one and I missed this :smile:

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/block.json
Comment (ObliviousHarmony) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/block.json:1]: I also double-checked that block to make sure it isn't being used anywhere. I don't _think_ it is as far as I can tell.

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx
Comment (imanish003) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx:1]: No worries, If it helps you may also take a look at https://github.com/woocommerce/woocommerce/pull/43314, In which I introduced this unique ID 🙂

File: pnpm-lock.yaml
Comment (imanish003) [inline pnpm-lock.yaml:1]: Regarding my earlier comment https://github.com/woocommerce/woocommerce/pull/44625#discussion_r1492032385, looks like `uuid` dependencies are still there in `plugins/woocommerce-blocks/package.json` file. May I know if that's intentional? 🙂

**Review by imanish003 — CHANGES_REQUESTED** [review]
Thanks for making the changes. I left one minor comment.

File: pnpm-lock.yaml
Comment (ObliviousHarmony) [reply pnpm-lock.yaml:1]: Funny story, it wasn't 😅 I guess I committed the changes to the lock file but not the changes to the `package.json` file!

Comment (imanish003) [reply pnpm-lock.yaml:1]: Haha No worries 😀

**Review by imanish003 — APPROVED** [review]
@ObliviousHarmony Thanks for making the changes. There are 2 more places from which we will need to remove the ID: [here](https://github.com/woocommerce/woocommerce/blob/78f1c2fa36e1991422d68290625575835d82b223/plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx#L31) & [here](https://github.com/woocommerce/woocommerce/blob/78f1c2fa36e1991422d68290625575835d82b223/plugins/woocommerce-blocks/assets/js/blocks/product-collection/constants.ts#L74).

I am preapproving the PR as everything else looks good to me :shipit:

**Comment by ObliviousHarmony** [comment]
I'm seeing this test failure in other executions of the same workflow and so I'm going to go ahead and merge anyway.

**Pull request merged**